backend/src/middleware/metricsMiddleware.ts

```
@@ -0,0 +1,36 @@
+
+/**
+ * Middleware to collect HTTP request metrics
+ */
+export function metricsMiddleware(
+  req: Request,
+  res: Response,
+  next: NextFunction
+): void {
+  // Skip if metrics disabled or path should be skipped
+  if (!isMetricsEnabled() || SKIP_PATHS.some((p) => req.path.startsWith(p))) {
+    return next();
```

Biome (2.1.2)

[error] The function should not return a value because its return type is void.

The function is here:

'void' signals the absence of value. The returned value is likely to be ignored by the caller.

(lint/correctness/noVoidTypeReturn)